src/lib.rs

```diff
@@ -180,14 +179,25 @@ fn build_poi(row: Row) -> Option<Poi> {
     let id = row.get("id");
     let name: String = row.get("name");
     let class: String = row.get("class");
-    let lat = row
+    let lat: f64 = row
         .get_opt("lat")?
         .map_err(|e| warn!("impossible to get lat for {} because {}", name, e))
         .ok()?;
-    let lon = row
+    let lon: f64 = row
```

Comment:
yes, it seems the compiler cannot infer type when calling `is_nan` before initializing `Coord`.  

Actually it would be less specific to ensure [`poi_coord.is_valid()`](https://github.com/CanalTP/mimirsbrunn/blob/master/libs/mimir/src/objects.rs#L566) instead.